src/openrct2/core/DataSerialiserTraits.h

```diff
@@ -749,22 +749,24 @@ template<> struct DataSerializerTraits_t<TrackDesignSceneryElement>
     }
 };
 
-template<> struct DataSerializerTraits_t<rct_vehicle_colour>
+template<> struct DataSerializerTraits_t<VehicleColour>
 {
-    static void encode(OpenRCT2::IStream* stream, const rct_vehicle_colour& val)
+    static void encode(OpenRCT2::IStream* stream, const VehicleColour& val)
     {
-        stream->Write(&val.body_colour);
-        stream->Write(&val.trim_colour);
+        stream->Write(&val.Body);
+        stream->Write(&val.Trim);
+        stream->Write(&val.Ternary);
     }
-    static void decode(OpenRCT2::IStream* stream, rct_vehicle_colour& val)
+    static void decode(OpenRCT2::IStream* stream, VehicleColour& val)
     {
-        stream->Read(&val.body_colour);
-        stream->Read(&val.trim_colour);
+        stream->Read(&val.Body);
+        stream->Read(&val.Trim);
+        stream->Read(&val.Ternary);
     }
-    static void log(OpenRCT2::IStream* stream, const rct_vehicle_colour& val)
+    static void log(OpenRCT2::IStream* stream, const VehicleColour& val)
     {
         char msg[128] = {};
-        snprintf(msg, sizeof(msg), "rct_vehicle_colour(body_colour = %d, trim_colour = %d)", val.body_colour, val.trim_colour);
+        snprintf(msg, sizeof(msg), "VehicleColour(body = %d, trim = %d, tertiary = %d)", val.Body, val.Trim, val.Ternary);
```

Comment:
Ternary or tertiary?

Reply:
> Ternary: Made up of three things; treble, triadic, triple, triplex
> Tertiary: Of third rank or order; subsequent. 

Tertiary seems better.

Reply:
Okay i've swapped it to tertiary. Annoyingly the plugin api uses ternary but its not the first nore the last where the plugin api uses an old name.